packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3369,7 +3360,7 @@ export class ContainerRuntime
 	private submitEnvelopedSignal(envelope: ISignalEnvelope, targetClientId?: string) {
 		const isBroadcastSignal = targetClientId === undefined;
 
-		if (isBroadcastSignal) {
+		if (isBroadcastSignal && this.connected) {
```

Comment:
Why is it okay to proceed to submitSignalFn if not connected?

We should also make sure to have test coverage of these states.

Reply:
I think this is because even if `this.connected` is false, we could still be connected to delta stream, so the signal can still technically be sent. The actual connection check is made at ConnectionManager.submitSignal, where we determine if the signal can be sent or not. 

```typescript
/**
* We want to lie to the ContainerRuntime when we are in readonly mode to prevent issues with pending
* ops getting through to the DeltaManager.
* The ContainerRuntime's "connected" state simply means it is ok to send ops
* See https://dev.azure.com/fluidframework/internal/_workitems/edit/1246
*/
```
https://github.com/microsoft/FluidFramework/blob/0637dcc29daaf0c39f0b99021e6a39d60b388dd9/packages/loader/container-loader/src/container.ts#L2483C1-L2486C77

Reply:
If the signal could be sent, then why wouldn't we track it?

Reply:
My thought process is that we know `this.connected === true` is the condition in which we 100% expect the signal to come back (i.e. if this.connected and we don't receive the signal back we have reason to submit and error event related to a dropped signal). 

When `this.connected === false` we can't expect anything since the status of our connection to the delta steam is unknown. We could be tracking signals sent while we are disconnected. Even if they are sent, we can't anticipate at the runtime level if we expect them to return.

Reply:
If we are disconnected at `ConnectionManager.submitSignal`, then won't we go through process of reestablishing the connection and resetting the tracking information?

FWIW we'll see `submitSignalDisconnected` error event if we send a signal while disconnected at ConnectionManager.

Reply:
Yes, if we're disconnected at `ConnectionManager.submitSignal`/if there's a `submitSignalDisconnected` error event we will it will always be followed by a SignalLost event. Currently, if we've been disconnected, the `trackingSignalSequenceNumber` is set to undefined. `trackingSignalSequenceNumber` will then be set again once the next signal is sent even if we are disconnected. So, if we send a signal while disconnected (`submitSignalDisconnected`) it means we introduced a gap in the signal sequence.

Reply:
Why will `submitSignalDisconnected` always be followed by `LostSignal`?
Perhaps some of the trouble is that the status reset doesn't happen on connect. It happens on disconnect / while disconnected.
I *think* the tracking will be better if everything is reset upon connect. In `setConnectionStateCore`, can we reset when connecting (possibly in addition to when disconnected/ing or exclusively) - check `changeOfState && connected` like we do for `flush()`. (To do both cases can check `if (!connected || changeOfState)`.)

We are going to need test coverage anyway. Perhaps the best approach here is to setup tests that put us in all of the situations.

Reply:
I changed so now we don't check `connected` on process or signal submit. We reset signal tracking data on reconnect now rather than disconnect. 

For testing, we have two main scenarios: (1) signals sent while connected but lost before processing due to a disconnect and (2) signals sent while disconnected. In both cases signal telemetry should not log any SignalLost events and should restart the expected signal sequence once reconnected.